indexing-service/src/main/java/org/apache/druid/indexing/overlord/ImmutableWorkerInfo.java

```diff
@@ -225,6 +244,13 @@ private int getWorkerParallelIndexCapacity(double parallelIndexTaskSlotRatio)
     return workerParallelIndexCapacity;
   }
 
+  public Map<String, Integer> incrementTypeSpecificCapacity(String type, int capacityToAdd)
```

Comment:
This method is a little misleading given that this is an immutable class.
I would advise simply inlining this change at the caller.

```
updatedMap = new HashMap(immutableWorkerInfo.getCurrCapacityUsedByTaskType()).merge(type, capacityToAdd, Integer::sum);
```